coops-ui/src/components/Common/OfficeAddresses.vue

```diff
@@ -0,0 +1,813 @@
+     */
+    private editAddress (): void {
+      // Check for inherited values
+      this.inheritDeliveryAddress = this.isSameAddress(this.mailingAddress, this.deliveryAddress)
+      this.inheritRecDeliveryAddress = this.isSameAddress(this.recMailingAddress, this.recDeliveryAddress)
+
+      // Populate the appropriate editable fields with temperorary data
+      this.deliveryAddressTemp = { ...this.deliveryAddress }
+      this.mailingAddressTemp = { ...this.mailingAddress }
+      this.recDeliveryAddressTemp = { ...this.recDeliveryAddress }
+      this.recMailingAddressTemp = { ...this.recMailingAddress }
+      this.showAddressForm = true
+    }
+
+    /**
+     * Cancels the editing of addresses, setting the addresses to the value they had before editing began.
+     */
+    private cancelEditAddress (): void {
+      this.deliveryAddress = { ...this.deliveryAddressTemp }
+      this.mailingAddress = { ...this.mailingAddressTemp }
+      this.recDeliveryAddress = { ...this.recDeliveryAddressTemp }
+      this.recMailingAddress = { ...this.recMailingAddressTemp }
+      this.showAddressForm = false
+    }
+
+    /**
+     * Updates the address data using what was entered on the forms.
+     */
+    private updateAddress (): void {
+      // Inherit the mailing address for delivery address for Registered Office Addresses
+      if (this.inheritDeliveryAddress) {
+        this.mailingAddress = { ...this.deliveryAddress }
+      }
+      // Inherit the mailing address from delivery address for Records Office Addresses
+      if (this.inheritRecDeliveryAddress) {
+        this.recMailingAddress = { ...this.recDeliveryAddress }
+      }
+      // Inherit the Records Office addresses from Registered Office Addresses
+      if (this.inheritRegisteredAddress) {
+        this.recDeliveryAddress = { ...this.deliveryAddress }
+        this.recMailingAddress = { ...this.mailingAddress }
+      }
+
+      this.showAddressForm = false
+      this.emitAddresses()
+      this.emitModified()
+    }
+
+    /**
+     * Resets the address data to what it was before any edits were done.
+     */
+    private resetAddress (): void {
+      this.deliveryAddress = { ...this.deliveryAddressOriginal }
+      this.mailingAddress = { ...this.mailingAddressOriginal }
+      this.recDeliveryAddress = { ...this.recDeliveryAddressOriginal }
+      this.recMailingAddress = { ...this.recMailingAddressOriginal }
+      this.emitAddresses()
+      this.emitModified()
+    }
+    /**
+     * Compares two address objects while omitting specified properties from the comparison.
+     *
+     * @param addressA The first address to compare
+     * @param addressB The second address to compare
+     * @param prop The property to omit during the comparison
+     *
+     * @return boolean A boolean indicating a match of addresses
+     */
+    private isSameWithoutProp (addressA: AddressIF, addressB: AddressIF, ...prop: Array<string>): boolean {
```

Comment:
There's a similar function in a mixin. Can we please have a single method for this?

Reply:
Great point, i did a significant refactor. I removed the address mixin, and created a method in the common mixin that can compare any objects against each other, whilst having the option to pass in a property to omit during the comparison.    I did this because the address comparison was only specific to addresses by name only.

Reply:
Yes, nicely done. I like it!